File: c/examples/ranger_read.c
/******************************************************************************
     * File: ranger_read.c
     * Description: Пример работы с датчиком расстояния. Цикличный опрос.
******************************************************************************/
#include "Ranger.h"
#include "bus_handler.h"

//Структура для записи полученных значений с датчика
struct sensor_data data;

//Адрес датчика. По умолчанию 1
int sensor_addr = 1;

int main()
{

  //Инициализация последовательного порта для работы с датчиком. По умолчанию /dev/RS_485
  if (init_port("/dev/RS_485",false) < 0)
  {
    return -1;
  }


  while(1)
  {

    //Отправка команды на измерение расстояния
    trig_sensor(sensor_addr);

    //О-идание отра-ения сигналов измерения
    usleep(10000);

    //Чтение измеренных значений расстояния
    data = read_sensor(sensor_addr);

    //Вывод измеренных расстояний
    printf("Data from sensor: \n");  
    printf("US_diatance: %i\n",data.US_distance);
    printf("IR_distance: %i\n\n",data.IR_distance);

  }

  return 0; 
}

File: python/examples/ranger_read.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""
ranger_read.py

Пример подключения к датчику расстояния и чтения с него данных.

"""
import Ranger
import time

#Порт шины данных. По умолчпнию /dev/RS_485
port        = '/dev/RS_485'

#Адрес датчика расстояния. По умолчанию 1
sensor_id   = 1

#Инициализация датчика
sensor = Ranger.Sensor(port,sensor_id,debug = True)

#Установка значения измеренной дистанции (мм), ниже которого сработает световая индикация. ЗЕЛЕНЫЙ - ИК датчик, СИНИЙ - УЗ датчик
sensor.set_min_dist(800) 

if __name__ == '__main__':

    while(True):

        #Отправка команды для измерения раастояния
        sensor.trig_sensor()

        time.sleep(0.1)

        #Чтение полученных данных о расстоянии
        vals = sensor.read_sensor()
        print('Data from sensor: {0}'.format(sensor_id))
        print('------------')
        for i in vals:
            print('{0} : {1}'.format(i, vals[i]))
        print('\n' * 3)


